pkg/operator/operator.go

```diff
@@ -85,46 +85,36 @@ func NewOpenShiftControllerManagerOperator(
 }
 
 func (c OpenShiftControllerManagerOperator) sync() error {
-	operatorConfig, err := c.operatorConfigClient.OpenShiftControllerManagers().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	cm := c.operatorConfigClient.OpenShiftControllerManagers()
+	operatorConfig, err := cm.Get(context.TODO(), "cluster", metav1.GetOptions{})
 	if err != nil {
 		return err
 	}
-	// manage status
+
 	originalOperatorConfig := operatorConfig.DeepCopy()
-	reasonString := ""
-	messageString := ""
 	switch operatorConfig.Spec.ManagementState {
 	case operatorapiv1.Removed:
 		fallthrough
-		// we equally do not allow removed/unmanaged
 	case operatorapiv1.Unmanaged:
-		reasonString = fmt.Sprintf("%sUnsupported", string(operatorConfig.Spec.ManagementState))
-		messageString = fmt.Sprintf("the controller manager spec was set to %s state, but that is unsupported, and has no effect on this condition", string(operatorConfig.Spec.ManagementState))
-
-		// as we are ignoring / not supporting unmanaged/removed, we still process any other inputs to the sync/reconciliation
-		// unlike what we used to do
+		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
+			Message: fmt.Sprintf("the controller manager spec is set to %s state", operatorConfig.Spec.ManagementState),
+			Reason:  fmt.Sprintf("%sUnsupported", string(operatorConfig.Spec.ManagementState)),
+			Status:  operatorapiv1.ConditionTrue,
+			Type:    workloadDegradedCondition,
+		})
+		if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
+			if _, err = cm.UpdateStatus(context.TODO(), operatorConfig, metav1.UpdateOptions{}); err != nil {
+				return err
+			}
+		}
 	case operatorapiv1.Managed:
-		// we want to empty out the reason/message string if transitioning from the other phases so the default setting above is good
-	}
-
-	for _, condition := range operatorConfig.Status.Conditions {
-		// do not change the current status as part of noting that unmanaged/removed are not supported
-		condition.Reason = reasonString
-		condition.Message = messageString
-		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, condition)
-	}
-	if !equality.Semantic.DeepEqual(operatorConfig.Status, originalOperatorConfig.Status) {
-		if _, err := c.operatorConfigClient.OpenShiftControllerManagers().UpdateStatus(context.TODO(), operatorConfig, metav1.UpdateOptions{}); err != nil {
-			return err
+		forceRequeue, err := syncOpenShiftControllerManager_v311_00_to_latest(c, operatorConfig)
+		if forceRequeue && err != nil {
+			c.queue.AddRateLimited(workQueueKey)
```

Comment:
why would we not empty out the conditions when managed and we did not get an error ?

Reply:
In the original changes, we were getting this error back:
```
E0730 08:02:46.640390       1 operator.go:164] key failed with : Operation cannot be fulfilled on openshiftcontrollermanagers.operator.openshift.io "cluster": the object has been modified; please apply your changes to the latest version and try again
```

With the changes in this PR the error is gone, and we add the following condition:

```
$ oc --namespace="openshift-controller-manager-operator" get openshiftcontrollermanagers.operator.openshift.io cluster -o json |jq '.status.conditions[-1]'
{
  "lastTransitionTime": "2020-07-30T08:27:41Z",
  "message": "the controller manager spec is set to Unmanaged state",
  "reason": "UnmanagedUnsupported",
  "status": "True",
  "type": "WorkloadDegraded"
}
```

Reply:
what happens when you switch from unmanaged back to managed?

Reply:
@gabemontero I misunderstood the use of status conditions in the operator, thanks for pointing this out. In the latest commit I'm reverting to the original behavior, please consider.